StanfordHCI/foundry
# Rk duplicate event

This PR consists of a functionality to duplicate event through  new config Icon added at the bottom left of the event tile. Drop-down consists of options for Duplicate, Edit, Delete. Duplicating event will create one more event right below it with exactly same property of its parent, Edit event would open dialog box to edit the event and Delete event would open a confirmation box to confirm deletion. 

There is a change in way events are created. Will need that to be tested thoroughly 

TESTING INSTRUCTIONS:

1) On the bottom left of event (Next to number of members), there should be a settings icon. Similar to https://thenounproject.com/term/settings/89645/ .  This should appear on all events (old and new) 

2) On click the Icon a drop-down should pop up at the mouse pointer,  it consists of 3 options. "Duplicate", "Edit", "Delete"

3) On selecting Duplicate form drop-down, a new event with exactly similar attributes of its parents should be created right below the old event(event on which drop down duplicate was selected).  Attributes e.g.: Event Start Hrs, Total Runtime, Project Coordinator, Members, Description etc. 

4) On selecting Edit, A dialog box to Edit the event should pop up. This should be exactly similar to the Edit event box set earlier. 

5) On selecting "Delete", A dialog box should appear asking for Confirmation to delete the selected event. If user selects 'Yes' on  confirm box, it should delete the event however selecting 'No' should just close the confirmation box.
